src/pocketmine/network/mcpe/protocol/LoginPacket.php

```diff
@@ -196,7 +199,9 @@ protected function decodeConnectionRequest() : void{
 			),
 			$this->clientData[self::I_LANGUAGE_CODE],
 			$this->extraData[self::I_XUID],
-			$this->clientData[self::I_CLIENT_RANDOM_ID]
+			$this->clientData[self::I_CLIENT_RANDOM_ID],
+			$this->clientData[self::I_DEVICE_MODEL],
+			$this->clientData[self::I_DEVICE_OS]
```

Comment:
These fields are not checked for presence, and it's possible for them to be omitted (for example by a custom client). If you want them to be mandatory, their presence should be checked in the `Validator` above.

Reply:
I was thinking about that. I dont see think it should be mandatory and could maybe set it to null or "Unknown" if it doesnt exist, which would you suggest?

Reply:
https://github.com/pmmp/PocketMine-MP/pull/2812/files#diff-5527e13f8305c41ad658af11214531e0R186

Reply:
Is that supposed to link to a specific line because it doesnt

Reply:
https://github.com/pmmp/PocketMine-MP/blob/0dec932df0e3cc1b2d97a43d0d7db330ecbf47c1/src/pocketmine/network/mcpe/protocol/LoginPacket.php#L186